earth_enterprise/src/scons/khEnvironment.py

```diff
@@ -141,6 +142,31 @@ def WriteToFileStrfunc(file_name, strn):
   return 'WriteToFile(%s, %s)' % (file_name, strn)
 
 
+def StringExpandFileFunc(target, source, env):
```

Comment:
Should we check the passed arguments ? or they will be always valid ?

Reply:
What kind of check do you mean?  You would get an exception if you pass a non-existent source file, or if the template contains an invalid format, etc.  Is there something you think should be checked?

Reply:
target or/and source or/and env can be null ? or this would not happen ?

Reply:
If any of these is `None`, you'd get an exception.